### ISSUE TITLE
'updatedCredentialsNotification' does not exist in type

### ISSUE THREAD

**Issue by nkhine** [aws-samples/amazon-cognito-passwordless-auth#122]
Hello,
Thank you for this library, I have an issue when trying to build and synthasise the cdk stack
```
TSError: ⨯ Unable to compile TypeScript:
stack.ts:60:9 - error TS2322: Type '{ authenticatorsTableProps: { removalPolicy: cdk.RemovalPolicy.DESTROY; billingMode: cdk.aws_dynamodb.BillingMode.PAY_PER_REQUEST; }; relyingPartyName: string; allowedRelyingPartyIds: string[]; attestation: "none"; userVerification: "required"; updatedCredentialsNotification: { ...; }; }' is not assignable to type '{ relyingPartyName?: string | undefined; allowedRelyingPartyIds: string[]; attestation?: "direct" | "enterprise" | "indirect" | "none" | undefined; userVerification?: "discouraged" | "preferred" | "required" | undefined; ... 6 more ...; api?: { ...; } | undefined; }'.
  Object literal may only specify known properties, and 'updatedCredentialsNotification' does not exist in type '{ relyingPartyName?: string | undefined; allowedRelyingPartyIds: string[]; attestation?: "direct" | "enterprise" | "indirect" | "none" | undefined; userVerification?: "discouraged" | "preferred" | "required" | undefined; ... 6 more ...; api?: { ...; } | undefined; }'.

60         updatedCredentialsNotification: {
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/amazon-cognito-passwordless-auth/cdk.d.ts:56:9
    56         fido2?: {
               ~~~~~
    The expected type comes from property 'fido2' which is declared here on type '{ userPool?: UserPool | undefined; userPoolClients?: UserPoolClient[] | undefined; userPoolProps?: Partial<UserPoolProps> | undefined; ... 7 more ...; logLevel?: "DEBUG" | ... 2 more ... | undefined; }'
stack.ts:141:24 - error TS2339: Property 'fido2NotificationFn' does not exist on type 'Passwordless'.

141 if (stack.passwordless.fido2NotificationFn) {
                           ~~~~~~~~~~~~~~~~~~~
stack.ts:143:24 - error TS2339: Property 'fido2NotificationFn' does not exist on type 'Passwordless'.

143     stack.passwordless.fido2NotificationFn,
                           ~~~~~~~~~~~~~~~~~~~

    at createTSError (/home/khine/go/src/github.com/aws-samples/amazon-cognito-passwordless-auth/end-to-end-example/cdk/node_modules/ts-node/src/index.ts:859:12)
    at reportTSError (/home/khine/go/src/github.com/aws-samples/amazon-cognito-passwordless-auth/end-to-end-example/cdk/node_modules/ts-node/src/index.ts:863:19)
    at getOutput (/home/khine/go/src/github.com/aws-samples/amazon-cognito-passwordless-auth/end-to-end-example/cdk/node_modules/ts-node/src/index.ts:1077:36)
    at Object.compile (/home/khine/go/src/github.com/aws-samples/amazon-cognito-passwordless-auth/end-to-end-example/cdk/node_modules/ts-node/src/index.ts:1433:41)
    at Module.m._compile (/home/khine/go/src/github.com/aws-samples/amazon-cognito-passwordless-auth/end-to-end-example/cdk/node_modules/ts-node/src/index.ts:1617:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1287:10)
    at Object.require.extensions.<computed> [as .ts] (/home/khine/go/src/github.com/aws-samples/amazon-cognito-passwordless-auth/end-to-end-example/cdk/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Function.Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12) {
  diagnosticCodes: [ 2322, 2339, 2339 ]
}

Subprocess exited with error 1
```

Looking at `node_modules/amazon-cognito-passwordless-auth/cdk.d.ts` is missing the `updatedCredentialsNotification` property

```
export declare class Passwordless extends Construct {
    userPool: cdk.aws_cognito.UserPool;
    userPoolClients?: cdk.aws_cognito.UserPoolClient[];
    secretsTable?: cdk.aws_dynamodb.Table;
    authenticatorsTable?: cdk.aws_dynamodb.Table;
    kmsKey?: cdk.aws_kms.IKey;
    createAuthChallengeFn: cdk.aws_lambda.IFunction;
    verifyAuthChallengeResponseFn: cdk.aws_lambda.IFunction;
    defineAuthChallengeResponseFn: cdk.aws_lambda.IFunction;
    preSignUpFn?: cdk.aws_lambda.IFunction;
    preTokenGenerationFn?: cdk.aws_lambda.IFunction;
    fido2Fn?: cdk.aws_lambda.IFunction;
    fido2challengeFn?: cdk.aws_lambda.IFunction;
    fido2Api?: cdk.aws_apigateway.RestApi;
    fido2ApiWebACL?: cdk.aws_wafv2.CfnWebACL;
    constructor(scope: Construct, id: string, props: {
        /** Your existing User Pool, if you have one already. This User Pool will then be equipped for Passwordless: Lambda triggers will be added. If you don't provide an existing User Pool, one will be created for you */
        userPool?: cdk.aws_cognito.UserPool;
        /** Your existing User Pool Clients, if you have them already. If you don't provide an existing User Pool Client, one will be created for you */
        userPoolClients?: cdk.aws_cognito.UserPoolClient[];
        /** If you don't provide an existing User Pool, one will be created for you. Pass any properties you want for it, these will be merged with properties from this solution */
        userPoolProps?: Partial<cdk.aws_cognito.UserPoolProps>;
        /** If you don't provide an existing User Pool Client, one will be created for you. Pass any properties you want for it, these will be merged with properties from this solution */
        userPoolClientProps?: Partial<cdk.aws_cognito.UserPoolClientOptions>;
        /**
         * The origins where you will be hosting your Web app on: scheme, hostname, and optionally port.
         * Do not include path as it will be ignored. The wildcard (*) is not supported.
         *
         * Example value: https://subdomain.example.org
         *
         * This property is required when using FIDO2 or Magic Links:
         * - For FIDO2 it is validated that the clientData.origin matches one of the allowedOrigins. Also, allowedOrigins is used as CORS origin setting on the FIDO2 credentials API.
         * - For Magic Links it is validated that the redirectUri (without path) in each Magic Link matches one of the allowedOrigins.
         */
        allowedOrigins?: string[];
        /**
         * Enable sign-in with FIDO2 by providing this config object.
         */
        fido2?: {
            relyingPartyName?: string;
            allowedRelyingPartyIds: string[];
            attestation?: "direct" | "enterprise" | "indirect" | "none";
            userVerification?: "discouraged" | "preferred" | "required";
            authenticatorAttachment?: "cross-platform" | "platform";
            residentKey?: "discouraged" | "preferred" | "required";
            /** Timeouts (in milliseconds) */
            timeouts?: {
                credentialRegistration?: number;
                signIn?: number;
            };
            authenticatorsTableProps?: TableProps;
            exposeUserCredentialIDs?: boolean;
            /**
             * Should users who previously registered FIDO2 credentials be forced to sign in with FIDO2?
             * FIDO2 is a phishing resistant signInMethod. As long as other signInMethods are still available,
             * there is a risk of phishing to the user, e.g. an attacker might trick the user into revealing the magic link.
             * Set to `true` to disallow other custom signInMethods if the user has one or more FIDO2 credentials.
             * @default false
             */
            enforceFido2IfAvailable?: boolean;
            api?: {
                /**
                 * The throttling burst limit for the deployment stage: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-request-throttling.html
                 *
                 * @default 1000
                 */
                throttlingBurstLimit?: number;
                /**
                 * The throttling rate limit for the deployment stage: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-request-throttling.html
                 *
                 * @default 2000
                 */
                throttlingRateLimit?: number;
                /**
                 * Create a log role for API Gateway and add this to API Gateway account settings?
                 * Set to false if you have already set this up in your account and region,
                 * otherwise that config will be overwritten.
                 *
                 * @default true
                 */
                addCloudWatchLogsRoleAndAccountSetting?: boolean;
                /**
                 * Add a WAF Web ACL with rate limit rule to the API deployment stage? The included Web ACL will have 1 rule:
                 * rate limit incoming requests to max 100 per 5 minutes per IP address (based on X-Forwarded-For header).
                 * If you want to customize the Web ACL, set addWaf to false and add your own Web ACL instead.
                 *
                 * @default true
                 */
                addWaf?: boolean;
                /**
                 * The rate limit per unique IP (using X-Forwarded-For header) that AWS WAF will apply: https://docs.aws.amazon.com/waf/latest/developerguide/waf-rule-statement-type-rate-based-high-level-settings.html
                 *
                 * @default 100
                 */
                wafRateLimitPerIp?: number;
                /**
                 * Pass any properties you want for the AWS Lambda Rest Api created, these will be merged with properties from this solution
                 */
                restApiProps?: Partial<cdk.aws_apigateway.RestApiProps>;
            };
        };
        /**
         * Enable sign-in with Magic Links by providing this config object
         * Make sure you've moved out of the SES sandbox, otherwise you can only send few e-mails,
         * and only from and to verified e-mail addresses: https://docs.aws.amazon.com/ses/latest/dg/request-production-access.html
         */
         ```

To reproduce:
```
git clone https://github.com/aws-samples/amazon-cognito-passwordless-auth.git
cd amazon-cognito-passwordless-auth/end-to-end-example/cdk
npm install
npx cdk synth
```

**Comment by nkhine** [comment]
I see this was added in https://github.com/aws-samples/amazon-cognito-passwordless-auth/commit/b93b731eb0c57cb0f5974072ac675d1051e6c440#diff-1f71a2db32ac2919a7f02ce750934d2195dd8007a306e279751a924b6f8f8ff5 commit

**Comment by ottokruse** [comment]
We still need to publish to NPM. It escaped us that the end to end test would be broken without it.

We'll publish to NPM soon. For now just remove the updatedCredentialNotification from  the e2e stack (you probably did) and go ahead without that notification.

Thanks for the report!

**Comment by nkhine** [comment]
Thanks @ottokruse

**Comment by ottokruse** [comment]
FYI just published [v0.10.0](https://github.com/aws-samples/amazon-cognito-passwordless-auth/releases/tag/v0.10.0) which should resolve this

**Comment by nkhine** [comment]
Great thank you @ottokruse
